File: Climbing_Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        
        int n_2 = 0;
        int n_1 = 1;
      
        // Dynamic programming : dp[n] = dp[n-1] + dp[n-2]
        for (int i=0; i<n; i++){
            int temp = n_2;
            n_2 = n_1;
            n_1 = n_1 + temp;
        }
        return n_1;
    }
};

File: Climbing_Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        n_2 = 0
        n_1 = 1
        # Dynamic programming : dp[n] = dp[n-1] + dp[n-2]
        for i in range(n):
            n_1, n_2 = n_1 + n_2, n_1
        return n_1
